lib/src/components/AbstractDialog/AbstractDialog.tsx

```diff
@@ -62,108 +79,124 @@ const AbstractDialog = forwardRef<HTMLDialogElement, Props>((props, ref) => {
-            evt.preventDefault();
-            onCancel(createCancelEvent());
-          }
-          else if (!hasNativeModalSupport) {
-            // emulate cancel-on-esc behavior in browsers which don't do it natively
-            onCancel(createCancelEvent());
-          }
-        }
-      }
-    }
-  }
-
+  // Handle open and close logic.
   useEffect(function() {
     /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
-    const el = dialogRef.current!,
-        // HTML <dialog> elements are supposed to remember what element was focused before they were opened,
-        // and restore focus to that element when they are closed. No browsers appear to implement this currently,
-        // so we do it ourselves
-        previouslyFocusedEl = document.activeElement;
-
-    setDialogRefState(el);
+    const dialogEl = dialogRef.current!;
+    setDialogRefState(dialogEl);
 
-    // if the open attr isn't being explicitly controlled at all, or is being passed as explictly open, use the
-    // show(Modal) methods to open it, getting all the built-in browser behavior around focus
+    // Open Dialog
     if (open == null || open) {
+      previouslyFocusedElementRef.current = document.activeElement as HTMLElement;
+
+      // Use dialog method
+      // We do not use showModal because it moves
+      // the dialog to top-level and it overlaps
+      // popovers generated from password managers.
       if (hasNativeModalSupport) {
-        if (isModal) {
-          /*
-          * This will cause the document to become "blocked by the modal dialog"
-          * (https://html.spec.whatwg.org/multipage/interaction.html#blocked-by-a-modal-dialog)
-          * meaning that only the modal and its contents are interactable/focusable.
-          *
-          * Note: not supported in safari, which is why the conditional is here. Once safari gets
-          * it together, we should be able to take advantage of the "top layer" functionality built into
-          * the browser around modals to simplify the dialog styling around z-index handling
-          */
-
-          // https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1029#issuecomment-968299542
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          (el as any).showModal();
-        }
-        else {
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          (el as any).show();
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (dialogEl as any).show();
+      }
+      else { // Use attribute
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (dialogEl as any).open = true;
+      }
+
+      // Only focus on autofocus element when it is a modal.
+      if (isModal) {
+        const autofocusEl = dialogEl.querySelector('[autofocus]') as HTMLElement | null;
+
+        if (autofocusEl) {
+          autofocusEl.focus();
```

Comment:
> What happens if `autofocus` is not present? Wouldn't we still want to focus on the first focusable element?

Nice catch: fixed here: c4bdd428071a62e6598c94a007de53148493b9ef

Reply:
> > What happens if `autofocus` is not present? Wouldn't we still want to focus on the first focusable element?
> 
> Nice catch: fixed here: [c4bdd42](https://github.com/sonatype/sonatype-react-shared-components/commit/c4bdd428071a62e6598c94a007de53148493b9ef)

Should we focus on `autofocus` element only when it is a modal? I see that `NxDrawer` also uses the `AbstractDialog`. So in case of `NxDrawer`, do we ignore `autofocus`?

Reply:
> > > What happens if `autofocus` is not present? Wouldn't we still want to focus on the first focusable element?
> > 
> > 
> > Nice catch: fixed here: [c4bdd42](https://github.com/sonatype/sonatype-react-shared-components/commit/c4bdd428071a62e6598c94a007de53148493b9ef)
> 
> Should we focus on `autofocus` element only when it is a modal? I see that `NxDrawer` also uses the `AbstractDialog`. So in case of `NxDrawer`, do we ignore `autofocus`?

Yup! It should only focus on autofocus element when it is a modal, at least according to what I interpret from:
https://html.spec.whatwg.org/multipage/interactive-elements.html#dialog-focusing-steps